proxywasm/proxytest/wasmwrapper.go

```diff
@@ -314,171 +314,297 @@ func wasmBool(b bool) uint64 {
 	_, err := r.NewHostModuleBuilder("env").
+		// proxy_log logs a message at the given log_level.
+		// See https://github.com/proxy-wasm/spec/tree/master/abi-versions/vNEXT#proxy_log
 		NewFunctionBuilder().
-		WithFunc(func(ctx context.Context, mod api.Module, logLevel uint32, messageData uint32, messageSize uint32) uint32 {
-			messageDataPtr := wasmBytePtr(ctx, mod, messageData, messageSize)
+		WithParameterNames("log_level", "message_data", "message_size").
+		WithResultNames("call_result").
+		WithFunc(func(ctx context.Context, mod api.Module, logLevel, messageData, messageSize uint32) uint32 {
+			messageDataPtr := wasmBytePtr(mod, messageData, messageSize)
 			return uint32(internal.ProxyLog(internal.LogLevel(logLevel), messageDataPtr, int(messageSize)))
 		}).
 		Export("proxy_log").
+		// proxy_set_property sets a property value.
+		// See https://github.com/proxy-wasm/spec/tree/master/abi-versions/vNEXT#proxy_set_property
 		NewFunctionBuilder().
-		WithFunc(func(ctx context.Context, mod api.Module, pathData uint32, pathSize uint32, valueData uint32, valueSize uint32) uint32 {
-			pathDataPtr := wasmBytePtr(ctx, mod, pathData, pathSize)
-			valueDataPtr := wasmBytePtr(ctx, mod, valueData, valueSize)
+		WithParameterNames("property_path_data", "property_path_size", "property_value_data",
+			"property_value_size").
+		WithResultNames("call_result").
+		WithFunc(func(ctx context.Context, mod api.Module, pathData, pathSize, valueData, valueSize uint32) uint32 {
+			pathDataPtr := wasmBytePtr(mod, pathData, pathSize)
+			valueDataPtr := wasmBytePtr(mod, valueData, valueSize)
 			return uint32(internal.ProxySetProperty(pathDataPtr, int(pathSize), valueDataPtr, int(valueSize)))
 		}).
 		Export("proxy_set_property").
+		// proxy_get_property gets a property value.
+		// See https://github.com/proxy-wasm/spec/tree/master/abi-versions/vNEXT#proxy_get_property
 		NewFunctionBuilder().
-		WithFunc(func(ctx context.Context, mod api.Module, pathData uint32, pathSize uint32,
-			returnValueData uint32, returnValueSize uint32) uint32 {
-			pathDataPtr := wasmBytePtr(ctx, mod, pathData, pathSize)
+		WithParameterNames("property_path_data", "property_path_size", "return_property_value_data",
+			"return_property_value_size").
+		WithResultNames("call_result").
+		WithFunc(func(ctx context.Context, mod api.Module, pathData, pathSize, returnValueData,
+			returnValueSize uint32) uint32 {
+			pathDataPtr := wasmBytePtr(mod, pathData, pathSize)
 			var returnValueHostPtr *byte
 			var returnValueSizePtr int
 			ret := uint32(internal.ProxyGetProperty(pathDataPtr, int(pathSize), &returnValueHostPtr, &returnValueSizePtr))
 			copyBytesToWasm(ctx, mod, returnValueHostPtr, returnValueSizePtr, returnValueData, returnValueSize)
 			return ret
 		}).
 		Export("proxy_get_property").
+		// proxy_send_local_response sends an HTTP response without forwarding request to the upstream.
+		//
+		// Note: proxy-wasm spec calls this proxy_send_http_response. See
```

Comment:
I added notes like these as there are so very many inconsistencies, and at least having the comments prevents time lost wandering to see what envoy does vs spec 🤷